ParkingFinder/handlers/user_information.py

```
@@ -41,6 +50,92 @@ def get(self, user_id):
+
+        :param String user_id:
+        :raise: NotFound:
+        :raise: AssertionError
+        :return:
+        """
+        # set to default to avoid missingArgumentException
+        access_token = self.get_argument(name='access_token', default=None)
+        # TODO: token validation
+        try:
+            assert user_id
+            user = yield UserService.get_user_detail(
+                user_id=user_id
+            )
+            
+            # json.loads() for string or unicode
+            # json load() for file
+            request_body = json.loads(self.request.body)
+            activated_vehicle_plate = request_body.get('activated_vehicle')
+            new_vehicle = request_body.get('new_vehicle', False)
+            flow_check = True
+            if new_vehicle:
+                if new_vehicle['plate'] and new_vehicle['brand'] and new_vehicle['model'] \
+                        and new_vehicle['color'] and new_vehicle['year']:
+                    vehicle = Vehicle({
+                        'plate': new_vehicle['plate'],
+                        'brand': new_vehicle['brand'],
+                        'model': new_vehicle['model'],
+                        'color': new_vehicle['color'],
+                        'year': new_vehicle['year']
+                    })
+                    # since NotFound has been handled in get_user_detail() so no need to try here
+                    yield UserService.register_vehicle(user.user_id, vehicle)
+                else:
+                    flow_check = False
+                    self.set_status(httplib.BAD_REQUEST)
+                    self.write({
+                        'error': "BAD REQUEST new vehicle info is incomplete \n"
+                    })
+
+            if activated_vehicle_plate:
+                # same reason above
+                user = yield UserService.activate_vehicle(user.user_id, activated_vehicle_plate)
```

You should throw error in service level if there is something wrong rather than return a user.